homeassistant/components/myfitnesspal/sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Support for Myfitnesspal totals as sensors."""
+from datetime import date
+from homeassistant.const import MASS_GRAMS
+from homeassistant.helpers.entity import Entity
+
+# REQUIREMENTS = ['myfitnesspal==1.13.3']
+TOTALS = ["sodium", "carbohydrates", "calories", "fat", "sugar", "protein"]
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set the sensor platform."""
+    dev = []
+    for resource in TOTALS:
```

Comment:
instead of creating a new sensor instance for each resource, make a loop on the update() to populate a dict. We don't want to overfill the registry like this.

Reply:
Is that creating one sensor and then users would use something to pull out the individual items? Would that contradict the need for a config?

Reply:
no you still need a config. If you just group a fitnesspal user int one sensor it makes it easier for multiple users to exist and make it easier to create a card that maps to one sensor instead of having to find all fitnesspal sensors then seperating multiple accounts, atc. like the fitbit sensor does right now. its kind of a pain.

Reply:
Believe this is covered now.